### PR TITLE
CCE preliminaries for GH+CCE executable

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

This is a rebased version of #2323 that contains all of the commits except the one for the actual executable.
I added one commit that factored the metavariables for CharacteristicExtract into two structs, one of which contains information that will be common to the stand-alone CCE and the GH+CCE executables

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by kidder** [comment]
@Sizheng-Ma can you repeat your check of running CCE with this branch

File: tests/Unit/Time/Actions/Test_ChangeStepSize.cpp [hunk 1]
```diff
@@ -189,3 +195,21 @@ SPECTRE_TEST_CASE("Unit.Time.Actions.ChangeStepSize", "[Unit][Time][Actions]") {
           reject_step);
   }
 }
+
+// [[OutputRegex, is not registered]]
+SPECTRE_TEST_CASE("Unit.Time.Actions.ChangeStepSize.subset_error",
```
Comment (nilsdeppe) [inline tests/Unit/Time/Actions/Test_ChangeStepSize.cpp]: can this now do an exception test?

**Comment by Sizheng-Ma** [comment]
> @Sizheng-Ma can you repeat your check of running CCE with this branch

The test looks good to me.

**Review by nilsdeppe — COMMENTED** [review]
Just one suggestion, feel free to squash immediately. There's a lot of time stepping code here. @wthrowe would you be able to look at this PR or at least the time stepping parts of it?

**Review by wthrowe — DISMISSED** [review]
Looks good.

**Comment by kidder** [comment]
@nilsdeppe rebased and squashed in error test -> exception test

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**